Source/SocketIO/Client/SocketIOClient.swift

```diff
@@ -295,8 +295,14 @@ open class SocketIOClient : NSObject, SocketIOClientSpec {
     }
 
     func emit(_ data: [Any], ack: Int? = nil, binary: Bool = true, isAck: Bool = false, completion: (() -> ())? = nil) {
+        // wrap the completion handler so it always runs async via handlerQueue
+        let wrappedCompletion = {[weak self, completion] in
```

Comment:
You shouldn't need to put completion in the capture list.

Reply:
Fixed